service/create/service.go

```diff
@@ -89,45 +96,101 @@ type Service struct {
 	bootOnce sync.Once
 }
 
-// Boot starts the service
+// Boot starts the service and implements the watch for the certificate TPR.
 func (s *Service) Boot() {
 	s.bootOnce.Do(func() {
-		s.Config.Logger.Log("info", "Booted cert-operator")
-
-		cert := CertificateSpec{
-			ClusterID:  "cert-test",
-			CommonName: "api.cert-test.g8s.eu-west-1.aws.test.private.giantswarm.io",
-			IPSANs:     []string{"10.0.0.4", "10.0.0.5"},
-			AltNames: []string{
-				"kubernetes",
-				"kubernetes.default",
-				"kubernetes.default.svc",
-				"kubernetes.default.svc.cluster.local",
-			},
-			AllowBareDomains: true,
-			TTL:              "720h",
+		if err := s.createTPR(); err != nil {
+			panic(fmt.Sprintf("could not create cluster resource: %#v", err))
 		}
+		s.Config.Logger.Log("info", "successfully created third-party resource")
+
+		_, certInformer := cache.NewInformer(
+			s.newCertificateListWatch(),
+			&certificatetpr.CustomObject{},
+			resyncPeriod,
+			cache.ResourceEventHandlerFuncs{
+				AddFunc:    s.handleCertAdd,
```

Comment:
The relations would be more easy to map if we align the function names with the struct properties IMO. So `handleCertAdd` would become `addFunc`.